### ISSUE TITLE
About DF-VAE

### ISSUE THREAD

**Issue by NNNNAI** [EndlessSora/DeeperForensics-1.0#8]
Thank you for your work!

Do you intend to release the relevant code and training scripts about the DF-VAE?

**Comment by NNNNAI** [comment]
Btw, as you mentioned in the paper,you use the stacked hourglass networks to extract landmarks. But I found the oringal implementation in the "Stacked hourglass networks for human pose estimation" did not include the landmarks detection. Could you tell me which version of stack hourglass you used in the paper?  Thank you very much.

**Comment by NNNNAI** [comment]
May I ask whether the stacked hourglass of the structure extraction module in dfvae will be trained together, that is, whether the parameters of the stacked hourglass will undergo gradient back propagation.

**Comment by EndlessSora** [comment]
> May I ask whether the stacked hourglass of the structure extraction module in dfvae will be trained together, that is, whether the parameters of the stacked hourglass will undergo gradient back propagation.

The answer is no. The gradient is locked for the stacked hourglass module.

**Comment by NNNNAI** [comment]
Thank you for your quick answer! BTW, in appendix, you mentioned using kl loss as one of the training losses of dfvae. Does the kl loss here mean that the structure encoder and appearance encoder both do kl loss? Or do you concatenate the latent representations from the structure encoder and appearance encoder, and then predict the mean and variance from the concatenated latent representations to calculate the kl loss ?

**Comment by NNNNAI** [comment]
Could you further explain how the df-vae combined the structure imformation and appearance and  how did the mean and variance in this part of klloss come from? I reproduced your Disentangled module and used 2DFAN in the structure extraction module to extract landmarks information, but the effect has not been good, and even the appearance of a complete face cannot be generated.
